lib/route.js

```diff
@@ -310,7 +310,12 @@ function buildRouting (options) {
           for (const hook of lifecycleHooks) {
             const toSet = this[kHooks][hook]
               .concat(opts[hook] || [])
-              .map(h => h.bind(this))
+              .map(h => {
+                if (typeof h !== 'function') {
```

Comment:
Can you move the logic right after the `onRoute` hook execution.
In here, the `typeof` validation will run through again and again for the same function.

We only need to guarantee the newly added `function` to be valid.

Reply:
Got it, I have moved the logic to right after the `onRoute` hook execution in my latest commit.

Also added new test to pass the coverage check.